scapy/contrib/isotp.py

```diff
@@ -0,0 +1,348 @@
+                         extended_rx_addr=None,
+                         listen_only=False):
+        option_flags = CAN_ISOTP_DEFAULT_FLAGS
+        if extended_addr is not None:
+            option_flags = option_flags | CAN_ISOTP_EXTEND_ADDR
+        else:
+            extended_addr = CAN_ISOTP_DEFAULT_EXT_ADDRESS
+
+        if extended_rx_addr is not None:
+            option_flags = option_flags | CAN_ISOTP_RX_EXT_ADDR
+        else:
+            extended_rx_addr = CAN_ISOTP_DEFAULT_EXT_ADDRESS
+
+        if listen_only:
+            option_flags = option_flags | CAN_ISOTP_LISTEN_MODE
+
+        sock.setsockopt(SOL_CAN_ISOTP,
+                        CAN_ISOTP_OPTS,
+                        self.__build_can_isotp_options(
+                            flags=option_flags,
+                            ext_address=extended_addr,
+                            rx_ext_address=extended_rx_addr))
+
+    def __init__(self,
+                 iface=None,
+                 sid=0,
+                 did=0,
+                 extended_addr=None,
+                 extended_rx_addr=None,
+                 listen_only=False,
+                 basecls=ISOTP):
+        self.iface = conf.contribs['NativeCANSocket']['iface'] \
+            if iface is None else iface
+        self.ins = socket.socket(socket.PF_CAN, socket.SOCK_DGRAM, CAN_ISOTP)
+        self.__setOptionFlags(self.ins,
+                              extended_addr,
+                              extended_rx_addr,
+                              listen_only)
+
+        self.ins.setsockopt(SOL_CAN_ISOTP,
+                            CAN_ISOTP_RECV_FC,
+                            self.__build_can_isotp_fc_options())
+        self.ins.setsockopt(SOL_CAN_ISOTP,
+                            CAN_ISOTP_LL_OPTS,
+                            self.__build_can_isotp_ll_options())
+
+        self.__bindSocket(self.ins, iface, sid, did)
+        self.outs = self.ins
+        if basecls is None:
+            warning('Provide a basecls ')
+        self.basecls = basecls
+
+    def recv(self, x=0xffff):
+        try:
+            pkt, sa_ll = self.ins.recvfrom(x)
+        except BlockingIOError:         # noqa: F821
+            warning('Captured no data, socket in non-blocking mode.')
+            return None
+        except socket.timeout:
+            warning('Captured no data, socket read timed out.')
+            return None
+        except OSError:
+            # something bad happened (e.g. the interface went down)
+            warning("Captured no data.")
+            return None
+
+        q = self.basecls(pkt)
+        q.time = get_last_packet_timestamp(self.ins)
+        return q
+
+    def send(self, x):
+        if hasattr(x, "sent_time"):
+            x.sent_time = time.time()
+        return self.outs.send(bytes(x))
+
+    def sr(self, *args, **kargs):
+        return sndrcv(self, *args, **kargs)
+
+    def sr1(self, *args, **kargs):
+        a, b = sndrcv(self, *args, **kargs)
```

Comment:
If you do not need `b`, please use `a = sndrcv(self, *args, **kargs)`. Also, the variable should be named `data`, for example, rather than using a meaningless letter.